# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderCanvasRenderer.h
# Commit: bbfe23c61add
# Full Hash: bbfe23c61add07377daf80696fd6996c0463f5b6
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 8fcfac450ecc -r bbfe23c61add gfx/layers/wr/WebRenderCanvasRenderer.h
--- a/gfx/layers/wr/WebRenderCanvasRenderer.h	Mon Jun 08 20:13:01 2020 +0000
+++ b/gfx/layers/wr/WebRenderCanvasRenderer.h	Mon Jun 08 20:34:15 2020 +0000
@@ -19,15 +19,13 @@
   explicit WebRenderCanvasRenderer(RenderRootStateManager* aManager)
       : mManager(aManager) {}
 
-  void Initialize(const CanvasInitializeData& aData) override;
-
   CompositableForwarder* GetForwarder() override;
 
  protected:
   RenderRootStateManager* mManager;
 };
 
-class WebRenderCanvasRendererAsync : public WebRenderCanvasRenderer {
+class WebRenderCanvasRendererAsync final : public WebRenderCanvasRenderer {
  public:
   explicit WebRenderCanvasRendererAsync(RenderRootStateManager* aManager)
       : WebRenderCanvasRenderer(aManager) {}
@@ -37,11 +35,10 @@
     return this;
   }
 
-  void Initialize(const CanvasInitializeData& aData) override;
+  void Initialize(const CanvasRendererData& aData) override;
   bool CreateCompositable() override;
 
   void ClearCachedResources() override;
-  void Destroy() override;
 
   void UpdateCompositableClientForEmptyTransaction();
 